crates/viewer/re_time_panel/tests/time_panel_tests.rs

```diff
@@ -137,17 +162,129 @@ pub fn run_time_panel_filter_tests(filter_active: bool, query: &str, snapshot_na
     run_time_panel_and_save_snapshot(test_context, time_panel, snapshot_name);
 }
 
-fn run_time_panel_and_save_snapshot(
+// --
+
+/// This test focuses on various kinds of entities and ensures their representation in the tree is
+/// correct regardless of the selected timeline and current time.
+//TODO(ab): we should also test what happens when GC kicks in.
+#[test]
+pub fn test_various_entity_kinds_in_time_panel() {
+    TimePanel::ensure_registered_subscribers();
+
+    for timeline in ["timeline_a", "timeline_b"] {
+        for time in [0, 5, i64::MAX] {
+            let mut test_context = TestContext::default();
+
+            log_all_data(&mut test_context);
+
+            test_context
+                .recording_config
+                .time_ctrl
+                .write()
+                .set_timeline_and_time(Timeline::new(timeline, TimeType::Sequence), time);
+
+            test_context
+                .recording_config
+                .time_ctrl
+                .write()
+                .set_time_view(TimeView {
+                    min: 0.into(),
+                    time_spanned: 10.0,
+                });
+
+            let time_panel = TimePanel::default();
+
+            run_expanded_time_panel_and_save_snapshot(
+                test_context,
+                time_panel,
+                &format!("various_entity_kinds_{timeline}_{time}"),
+            );
+        }
+    }
+}
+
+pub fn log_all_data(test_context: &mut TestContext) {
```

Comment:
nit: a bit irritating that at the top there's some tests that log stuff and here we have a "log_all_data" which logs all the data for a _specific_ test